src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,32 @@
+import { TfiClose } from 'react-icons/tfi';
+
+type PropTypes = {
+  showModal: boolean;
+  onClose: () => void;
+  children: React.ReactNode;
+};
+
+// eslint-disable-next-line react/prop-types
+export const Modal: React.FC<PropTypes> = ({ showModal, onClose, children }) => {
+  const onCloseClick = (event: { stopPropagation: () => void }) => {
+    onClose();
+    event.stopPropagation();
+  };
+  return (
+    <section
+      className={`location-selector fixed inset-0 overflow-hidden w-full flex flex-col z-50 justify-center items-center 
+        transition-all ease-out duration-200 ${showModal ? 'visible bg-white' : 'invisible'}`}
+    >
+      <div
```

Comment:
I think that div should be a button because you when you click  - it is good to make it as a button